### PR TITLE
[External][Smoke] Adding the aomp smoke suite to externals

### PR DESCRIPTION
**Pull request by ajarmusch**
Connecting the tests from [ROCm-Developer-Tools](https://github.com/ROCm-Developer-Tools/aomp/tree/aomp-dev/test/smoke) to the External folder. For further testing of OpenMP Clang in LLVM.

### CODE REVIEW

File: External/smoke/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+    assume_no_openmp
+    atomic
+    atomic_double
+    big-jump-loop-1
+    big-jump-loop-collapse
+    big-jump-loop-env-vars
+    big-jump-loop-matmul
+    big-jump-loop-split-2
+    big-jump-loop-split-3
+    big-jump-loop-split-4
```
Comment (jplehr) [inline External/smoke/CMakeLists.txt:28]: These depend on downstream functionality and won't work with upstream. At least the filecheck in those test cases.

File: External/smoke/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+    bugz-50967
+    bugz-50967-c
+    bugz-50968
+    CanonNE
+    check_smoke.sh
```
Comment (jplehr) [inline External/smoke/CMakeLists.txt:33]: That is the driver used in our team to run the tests and check them for their results.

File: External/smoke/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+    complex
+    complex2
+    complex-debug
+    complex_reduction
+    const_global_ctor-308039
+    d2h_slow_copy
+    data_enter_issue01
+    data_issue_59
+    data_share1
+    data_share2
```
Comment (jplehr) [inline External/smoke/CMakeLists.txt:86]: This one for sure has a `printf` that is currently unsupported upstream, I believe.

File: External/smoke/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+    declare_target_pointer
+    decl_map_targ
+    devices
+    devito_prob1
+    double_complex_scalar
+    envvar-sync-async
+    envvar-sync-async-perf
```
Comment (jplehr) [inline External/smoke/CMakeLists.txt:93]: I think those two also depend on downstream functionality.

File: External/smoke/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,572 @@
...
+    nested_targ
+    noarg_target_function
+    no-loop-8
+    no-loop-sched
+    no-loop-split-1
+    no-loop-split-2
+    no-loop-split-3
+    no-loop-split-with-inner-private-clause
+    no-loop-with-continue
+    no-offload-arch
```
Comment (jplehr) [inline External/smoke/CMakeLists.txt:163]: These depend on downstream functionality.

File: External/smoke/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,572 @@
...
+    xteam-red-1
+    xteam-red-1-g
+    xteam-red-2
+    xteam-red-3
+    xteam-red-4
+    xteam-red-cmdline-option1
+    xteam-red-collapse
+    xteam-red-teams
+    xteam-red-threads
+    xteam-red-unsupported
```
Comment (jplehr) [inline External/smoke/CMakeLists.txt:243]: These depend on downstream functionality.

**Review by jplehr — COMMENTED** [review]
I have left an initial comment about tests that for sure depend on downstream functionality. There may be more for which this is true.

Also, as mentioned in Slack, you probably want to pin the version you actually pulling, as we may move tests from `smoke` into `smoke-limbo` or `smoke-fails`.

A general question: Do you intend to use the `check_smoke.sh` script run them?

File: External/smoke/CMakeLists.txt (hunk 2 above)
Comment (ajarmusch) [reply External/smoke/CMakeLists.txt:33]: Removed all the check scripts in commit #6c88027f

File: External/smoke/CMakeLists.txt (hunk 3 above)
Comment (ajarmusch) [reply External/smoke/CMakeLists.txt:86]: is this only for data_share2 or lines 83 to 86?

File: External/smoke/CMakeLists.txt (hunk 1 above)
Comment (ajarmusch) [reply External/smoke/CMakeLists.txt]: updated with: cb9b0dbd31eea10325286bf389d0cac20cf33260

File: External/smoke/CMakeLists.txt (hunk 4 above)
Comment (ajarmusch) [reply External/smoke/CMakeLists.txt]: updated with: cb9b0dbd31eea10325286bf389d0cac20cf33260

File: External/smoke/CMakeLists.txt (hunk 5 above)
Comment (ajarmusch) [reply External/smoke/CMakeLists.txt]: updated with: cb9b0dbd31eea10325286bf389d0cac20cf33260

File: External/smoke/CMakeLists.txt (hunk 6 above)
Comment (ajarmusch) [reply External/smoke/CMakeLists.txt]: updated with: cb9b0dbd31eea10325286bf389d0cac20cf33260

**Review by fel-cab — APPROVED** [review]

**Pull request merged**